E-Commerce/readme.md

```diff
@@ -46,13 +46,101 @@ erDiagram
 Write SQL queries to complete the following tasks:
 
 - [ ] List all the products whose name contains the word "socks"
+
+SELECT * 
+FROM products
+WHERE product_name LIKE '%socks%';
+
+
+
 - [ ] List all the products which cost more than 100 showing product id, name, unit price, and supplier id
+
+ SELECT p.id, p.product_name, pa.unit_price, pa.supp_id
+ FROM products AS p
+ INNER JOIN product_availability AS pa
+ ON p.id = pa.prod_id
+ WHERE pa.unit_price > 100;
+
+
+
+
 - [ ] List the 5 most expensive products
+
+SELECT p.id, p.product_name, pa.unit_price
+ FROM products AS p
+ INNER JOIN product_availability AS pa
+ ON p.id = pa.prod_id
+ ORDER BY pa.unit_price DESC 
+ LIMIT 5;
+
+
+
+
 - [ ] List all the products sold by suppliers based in the United Kingdom. The result should only contain the columns product_name and supplier_name
+
+SELECT p.product_name, s.supplier_name
+FROM products AS p
+INNER JOIN product_availability AS pa
+ON p.id = pa.prod_id
+INNER JOIN suppliers AS s
+ON pa.supp_id = s.id
+WHERE s.country = 'United Kingdom';
+
+
+
 - [ ] List all orders, including order items, from customer named Hope Crosby
+
+SELECT  
+    oi.product_id,
+    oi.order_id,
+    o.order_date,
+    o.order_reference,
+    o.customer_id,
+    c.name AS customer_name
+FROM orders AS o
+INNER JOIN order_items AS oi
+ON o.id = oi.order_id
+INNER JOIN customers AS c 
+ON o.customer_id = c.id
+WHERE c.name = 'Hope Crosby';
+
+
+
 - [ ] List all the products in the order ORD006. The result should only contain the columns product_name, unit_price, and quantity
+
+SELECT
+    p.product_name,
+    pa.unit_price,
+    oi.quantity
+FROM products AS p
+INNER JOIN product_availability AS pa
+ON p.id = pa.prod_id
+INNER JOIN order_items AS oi 
+ON pa.prod_id = oi.product_id
+INNER JOIN orders AS o
+ON oi.order_id = o.id 
+WHERE o.order_reference = 'ORD006';
+
+
+
 - [ ] List all the products with their supplier for all orders of all customers. The result should only contain the columns name (from customer), order_reference, order_date, product_name, supplier_name, and quantity
 
```

Comment:
There are 59 rows in the result from this query, but if I have a look at the order_items table, there are only 19 rows there. That tells me something is off here, since we were trying to return information about products and suppliers in orders.
Hint: Does it make sense to use product_availability for this question?